Source/Controllers/DataControllers/SPTableData.m

```diff
@@ -442,75 +442,89 @@ - (BOOL) updateInformationForCurrentView
+        }
+
+        return nil;
+    }
 
-	// Retrieve the table syntax string
-	NSArray *syntaxResult = [theResult getRowAsArray];
-	NSArray *resultFieldNames = [theResult fieldNames];
+    // Retrieve the table syntax string
+    NSArray *syntaxResult = [theResult getRowAsArray];
+    NSArray *resultFieldNames = [theResult fieldNames];
+
+    // Only continue if syntaxResult is not nil. This accommodates causes where the above query caused the
+    // connection reconnect dialog to appear and the user chose to close the connection.
+    if (!syntaxResult) return nil;
+    
+    // A NULL value indicates that the user does not have permission to view the syntax
+    if ([[syntaxResult safeObjectAtIndex:1] isNSNull] || [syntaxResult safeObjectAtIndex:1] == nil) {
+        [NSAlert createWarningAlertWithTitle:NSLocalizedString(@"Permission Denied", @"Permission Denied") message:NSLocalizedString(@"The creation syntax could not be retrieved due to a permissions error.\n\nPlease check your user permissions with an administrator.", @"Create syntax permission denied detail") callback:nil];
+         
+         if (changeEncoding) [mySQLConnection restoreStoredEncoding];
+         return nil;
+    }
 
-	// Only continue if syntaxResult is not nil. This accommodates causes where the above query caused the
-	// connection reconnect dialog to appear and the user chose to close the connection.
-	if (!syntaxResult) return nil;
-	
-	// A NULL value indicates that the user does not have permission to view the syntax
-	if ([[syntaxResult safeObjectAtIndex:1] isNSNull] || [syntaxResult safeObjectAtIndex:1] == nil) {
-		[NSAlert createWarningAlertWithTitle:NSLocalizedString(@"Permission Denied", @"Permission Denied") message:NSLocalizedString(@"The creation syntax could not be retrieved due to a permissions error.\n\nPlease check your user permissions with an administrator.", @"Create syntax permission denied detail") callback:nil];
-		 
-		 if (changeEncoding) [mySQLConnection restoreStoredEncoding];
-		 return nil;
-	}
+    tableCreateSyntax = [[NSString alloc] initWithString:[syntaxResult objectAtIndex:1]];
+    
+    NSDictionary *tableData = [self parseCreateStatement:tableCreateSyntax ofType:[resultFieldNames objectAtIndex:0]];
+    
+    if (changeEncoding) [mySQLConnection restoreStoredEncoding];
 
-	tableCreateSyntax = [[NSString alloc] initWithString:[syntaxResult objectAtIndex:1]];
-	
-	NSDictionary *tableData = [self parseCreateStatement:tableCreateSyntax ofType:[resultFieldNames objectAtIndex:0]];
-	
-	if (changeEncoding) [mySQLConnection restoreStoredEncoding];
+    return tableData;
+}
 
-	return tableData;
+/**
+ * Retrieve the CREATE statement for a table/view and return extracted table
+ * structure information.
+ * @attention This method will interact with the UI on errors/connection loss!
+ */
+- (NSDictionary *) informationForTable:(NSString *)tableName
+{
+    return [self informationForTable:tableName fromDatabase:nil];
 }
 
```

Comment:
Does anything use this method? 

Either way, we may be able to kill off the shorthand one (And remove from header too) and just switch to the fromDatabase variant directly.

Reply:
This method is used in 6 different places. I can modify them by switching to the fromDatabase variant ?
Thus, there will be only one method. But that won't resole the `if` in case of the database parameter is nil.

Reply:
Let's do that! Combine to one method (get rid of the one without database short method), and for now leave the if block if the method which handles null database by not doing database. lookups. Then let's merge!

Reply:
it's done!